esp-hal/src/utils/macros.rs

```diff
@@ -0,0 +1,107 @@
+#[doc(hidden)]
```

Comment:
nitpick: I would probably drop the `utils` module, and inline `macros.rs` into the top level source

Reply:
i thought I'd move more here :D